### PR TITLE
Add MongoDB support

### PR DESCRIPTION
**Pull request by justinchuch**
Starting with the basics:

- `insert`
- `find`

### CODE REVIEW

File: driver-mongodb/src/main/java/io/nosqlbench/driver/mongodb/MongoAction.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @Override
+    public int runCycle(long cycleValue) {
+        ReadyMongoStatement rms;
+        Bson queryBson;
+        try (Timer.Context bindTime = activity.bindTimer.time()) {
+            rms = sequencer.get(cycleValue);
+            queryBson = rms.bind(cycleValue);
+
+            if (activity.isShowQuery()) {
+                logger.info("Query(cycle={}):\n{}", cycleValue, queryBson);
```
Comment (jshook) [inline driver-mongodb/src/main/java/io/nosqlbench/driver/mongodb/MongoAction.java:42]: We might want to add a warning on this when it is enabled and cycles>1000 total, similar to how we do with the CQL activity.

File: driver-mongodb/src/main/java/io/nosqlbench/driver/mongodb/MongoAction.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            try (Timer.Context executeTime = activity.executeTimer.time()) {
+                MongoDatabase database = activity.getDatabase();
+                ReadPreference readPreference = rms.getReadPreference();
+                Document resultDoc = database.runCommand(queryBson, readPreference);
+
+                double ok = resultDoc.getDouble("ok");
+                activity.resultSetSizeHisto.update(resultDoc.getInteger("n", 0));
+                return ok == 1.0d ? 0 : 1;
+            } catch (Exception e) {
+                logger.error("Failed to runCommand {} on cycle {}, tries {}", queryBson, cycleValue, i, e);
```
Comment (jshook) [inline driver-mongodb/src/main/java/io/nosqlbench/driver/mongodb/MongoAction.java:57]: As a future enhancement, we will want to capture total results and successful results as separate metrics, requiring a slight refactoring of the exception handling here.

File: driver-mongodb/src/main/resources/activities/mongodb-basic.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# nb -v run driver=mongodb yaml=mongodb-basic connection=mongodb://127.0.0.1 database=testdb tags=phase:rampup
```
Comment (jshook) [inline driver-mongodb/src/main/resources/activities/mongodb-basic.yaml:1]: There is now a `description` property supported in YAMLs that hasn't been documented yet. You might want to add a `description:` to this yaml.

File: driver-mongodb/src/main/resources/mongodb.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# MongoDB Driver
+
+This is a driver for MongoDB.
```
Comment (jshook) [inline driver-mongodb/src/main/resources/mongodb.md:3]: We should document all the parameters that are special to this activity type.

File: pom.xml [hunk 5]
```diff
@@ -68,6 +69,16 @@
                 <module>driver-cqld4</module>
             </modules>
         </profile>
+        <profile>
+            <id>with-mongodb</id>
+            <activation>
+                <activeByDefault>false</activeByDefault>
```
Comment (jshook) [inline pom.xml:75]: This needs to be set to true to work with the module inclusion above, in which case the lower module inclusion also applies. I suggest we just make it on by default now that it is building and working.

File: driver-mongodb/src/main/resources/activities/mongodb-basic.yaml (hunk 3 above)
Comment (justinchuch) [reply driver-mongodb/src/main/resources/activities/mongodb-basic.yaml:1]: Updated. :ok_hand:

File: pom.xml (hunk 5 above)
Comment (justinchuch) [reply pom.xml]: Updated. :ok_hand:

File: driver-mongodb/src/main/java/io/nosqlbench/driver/mongodb/MongoAction.java (hunk 2 above)
Comment (justinchuch) [reply driver-mongodb/src/main/java/io/nosqlbench/driver/mongodb/MongoAction.java:74]: Added `resultSuccess` metric, renamed the metric `execute` to `result`.

File: driver-mongodb/src/main/resources/mongodb.md (hunk 4 above)
Comment (justinchuch) [reply driver-mongodb/src/main/resources/mongodb.md]: Updated. :ok_hand:

**Pull request merged**